x/authority/types/authorizations_test.go

```diff
@@ -0,0 +1,196 @@
+package types_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+
+	"github.com/zeta-chain/zetacore/testutil/sample"
+	"github.com/zeta-chain/zetacore/x/authority/types"
+	crosschaintypes "github.com/zeta-chain/zetacore/x/crosschain/types"
+	fungibletypes "github.com/zeta-chain/zetacore/x/fungible/types"
+	lightclienttypes "github.com/zeta-chain/zetacore/x/lightclient/types"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+func TestAuthorizationList_SetAuthorizations(t *testing.T) {
```

Comment:
Would be better to make the assertions clear to use values defined in the test, where urls and policyTypes can clearly be seen instead of using `DefaultAuthorizationsList` or `sample`

`sample` is good when need to fill types to be used in test but we should manually define the values we want to assert

Reply:
In my opinion, the default and sample here make the number of lines of code shorter, so it's easier to read. 
Defining the same values again and again might be unnecessary. 

I am guessing you mean creating  list like this in every function 
https://github.com/zeta-chain/zeta-node/blob/e81d37068a35f8a0f3819c0dc6f2a6ab9c0fc965/x/authority/types/authorizations_test.go#L81-L90
Please correct me if you were referring to something else ?

Reply:
Test should be clear and efficient, not short.

We had issues in the past of missing cases to check all conditions and in this case TBT makes much easier to see what is being tested, if edge case are missing

I don't think this makes the tests less clear. In you example above we have long lines and usage of sample and default which makes harder to check what is being checked.
If we can use literal values and use new lines for every elements this would make easy to visualize
```
list: {
  {
    url: "foo"
    policy: "operation"
  }
}
newElement: {
  url: "bar"
  policy: "emergency"
}
newList: {
  {
    url: "foo"
    policy: "operation"
  },
  {
    url: "bar"
    policy: "emergency"
  }
}
```
Also in the current test, we don't check that the other elements are left untouched, and with `CheckAuthorizationExists` we don't assert the current policy types.

Finally, TDT allows to easily add new cases so you can check edge cases, here for example we don't check when setting to anempty array

Reply:
4b198d44901073a5d21399510efc92943d7c01e6